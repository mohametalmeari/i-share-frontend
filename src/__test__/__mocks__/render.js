import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../redux/store';

const customRender = (ui, options) => render(ui, {
  wrapper: ({ children }) => <Provider store={store}>{children}</Provider>,
  ...options,
});

export default customRender;