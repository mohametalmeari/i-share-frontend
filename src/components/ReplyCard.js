import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { deleteComment } from '../redux/posts/commentSlice';

const ReplyCard = ({
  id, photoId, commentId, name, content, likes, liked, control,
}) => {
  const dispatch = useDispatch();
  const handleDelete = async () => {
    await dispatch(deleteComment({ id, photoId, commentId }));
  };

  return (
    <div>
      {id}
      ,
      {name}
      :
      {content}
      ,
      {likes}
      ,
      {liked}

      {control && (
      <button type="button" onClick={handleDelete}>
        Delete reply
      </button>
      )}
      <br />
    </div>
  );
};

export default ReplyCard;

ReplyCard.propTypes = {
  id: PropTypes.number.isRequired,
  photoId: PropTypes.number.isRequired,
  commentId: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired,
  likes: PropTypes.number.isRequired,
  liked: PropTypes.bool.isRequired,
  control: PropTypes.bool.isRequired,
};
