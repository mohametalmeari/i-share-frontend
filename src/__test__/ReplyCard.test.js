import render from 'render';
import ReplyCard from '../components/ReplyCard';

describe(ReplyCard, () => {
  const reply = {
    id: 123,
    user: { name: 'Alex', control: false },
    content: 'Test reply',
    likes: 7,
    liked: true,

  };
  const photoId = 456;
  const id = 789;
  const refreshLikes = () => {};
  const removeFromReplies = () => {};

  test('renders Snapshot', () => {
    const { container } = render(
      <ReplyCard
        id={reply.id}
        photoId={photoId}
        commentId={id}
        name={reply.user.name}
        content={reply.content}
        likes={reply.likes}
        liked={reply.liked}
        control={reply.user.control}
        refreshLikes={refreshLikes}
        removeFromReplies={removeFromReplies}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
