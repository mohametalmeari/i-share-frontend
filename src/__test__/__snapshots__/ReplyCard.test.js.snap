// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReplyCard renders Snapshot 1`] = `
<div>
  <div>
    <div>
      <b>
        Alex
        : 
      </b>
      <span>
        Test reply
      </span>
    </div>
    <div
      class="interaction-counts"
    >
      <span>
        7 Likes
      </span>
    </div>
    <div
      class="comment-interact-container"
    >
      <button
        class="link comment-interact"
        type="button"
      >
        Unlike
      </button>
    </div>
  </div>
</div>
`;
